cranelift/codegen/src/isa/x64/lower.rs

```diff
@@ -899,45 +899,76 @@ fn lower_insn_to_regs<C: LowerCtx<I = Inst>>(
             let lhs = input_to_reg_mem(ctx, inputs[0]);
             let rhs = input_to_reg(ctx, inputs[1]);
             let dst = output_to_reg(ctx, outputs[0]);
+            let ty = ty.unwrap();
+
+            // Move the value to the correct register. TODO Why? This seems unnecessary and potentially incorrect--at best it should be explained here.
+            let mov_op = match ty {
+                types::F32 => SseOpcode::Movss,
+                types::F64 => SseOpcode::Movsd,
+                types::F32X4 | types::F64X2 => SseOpcode::Movaps,
+                _ => panic!(
+                    "invalid type: expected one of [F32, F64, F32X4, F64X2], found {}",
+                    ty
+                ),
+            };
+            ctx.emit(Inst::xmm_mov(mov_op, lhs, dst, None));
```

Comment:
pre-existing: can you use `Inst::gen_move` here, instead? (and fill in the right bits in `gen_move` to support vector instructions)?